src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,151 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.*;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj.controller.LinearQuadraticRegulator;
+import edu.wpi.first.wpilibj.estimator.KalmanFilter;
+import edu.wpi.first.wpilibj.system.LinearSystem;
+import edu.wpi.first.wpilibj.system.LinearSystemLoop;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpiutil.math.Matrix;
+import edu.wpi.first.wpiutil.math.Nat;
+import edu.wpi.first.wpiutil.math.VecBuilder;
+import edu.wpi.first.wpiutil.math.Vector;
+import edu.wpi.first.wpiutil.math.numbers.N1;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.subsystems.UnitModel;
+import frc.robot.utils.MovingAverage;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import static frc.robot.Constants.Shooter.*;
+
+/**
+ * The shooter class represents the physical shooter.
+ * The purpose of the flywheel is to provide a set of functions to handle a shooting situation in a game.
+ *
+ * @author Barel
+ * @version 1.0
+ * @using TalonFX
+ * @since 2021
+ */
+public class Shooter extends SubsystemBase {
+    private final TalonFX main = new TalonFX(Ports.Shooter.MAIN);
+    private final TalonFX aux = new TalonFX(Ports.Shooter.AUX);
+    private final UnitModel unitModel = new UnitModel(TICKS_PER_ROTATION);
+    private final MovingAverage velocityEstimator;
+
+    private final LinearSystemLoop<N1, N1, N1> stateSpacePredictor;
+
+    public Shooter() {
+        // Configure the motors
+        main.setInverted(Ports.Shooter.MAIN_INVERTED);
+        aux.setInverted(Ports.Shooter.AUX_INVERTED);
+        main.setSensorPhase(Ports.Shooter.IS_SENSOR_INVERTED);
+
+        main.setNeutralMode(NeutralMode.Coast);
+        aux.setNeutralMode(NeutralMode.Coast);
+
+        main.configForwardLimitSwitchSource(LimitSwitchSource.Deactivated, LimitSwitchNormal.Disabled);
+        main.configReverseLimitSwitchSource(LimitSwitchSource.Deactivated, LimitSwitchNormal.Disabled);
+
+        main.enableVoltageCompensation(true);
+        aux.enableVoltageCompensation(true);
+
+        main.configVoltageCompSaturation(Constants.NOMINAL_VOLTAGE);
+        aux.configVoltageCompSaturation(Constants.NOMINAL_VOLTAGE);
+
+        aux.follow(main);
+
+        this.stateSpacePredictor = constructLinearSystem();
+        this.velocityEstimator = new MovingAverage(readCSV());
+    }
+
+    private LinearSystemLoop<N1, N1, N1> constructLinearSystem() {
+        // https://file.tavsys.net/control/controls-engineering-in-frc.pdf Page 76
+        Vector<N1> A = VecBuilder.fill(-Math.pow(GEAR_RATIO, 2) * Kt / (Kv * OMEGA * J)); //Change the amount of cells and rows
```

Comment:
> Change the amount of cells and rows

WDYM?
Also, Grammarly says that "number" should be used instead of "amount".

Reply:
It's comment for myself, I need to delete it.

Reply:
That was my guess.